device/common/pddf/plugins/sfputil.py

```diff
@@ -0,0 +1,210 @@
+        port_ps = pddf_obj.get_path(device,"xcvr_lpmode")
+        #print "port_ps value is : %s"%port_ps
+        try:
+            reg_file = open(port_ps, 'w')
+        except IOError as e:
+            if not self.get_presence(port_num):
+                return False
+            if port_num not in self.qsfp_ports:
+                return False # Read from eeprom only for QSFP ports
+            try:
+                eeprom = None
+                eeprom = open(self.port_to_eeprom_mapping[port_num], "rb")
+                eeprom.seek(93)
+                lpmode = ord(eeprom.read(1))
+                
+                if ((lpmode & 0x3) == 0x3):
+                    return True # Low Power Mode if "Power override" bit is 1 and "Power set" bit is 1
+                else:
+                    return False # High Power Mode if one of the following conditions is matched:
+                                 # 1. "Power override" bit is 0
+                                 # 2. "Power override" bit is 1 and "Power set" bit is 0
+            except IOError as e:
+                print "Error: unable to open file: %s" % str(e)
+                return False
+            finally:
+                if eeprom is not None:
+                    eeprom.close()
+                    time.sleep(0.01)
+        #raise NotImplementedError
+
+    def set_low_power_mode(self, port_num, lpmode):
+        # Check for invalid port_num
+        if port_num < self._port_start or port_num > self._port_end:
+            return False
+
+        device = "PORT" + "%d"%(port_num+1)
+        port_ps = pddf_obj.get_path(device,"xcvr_lpmode")
+        try:
+            reg_file = open(port_ps, 'w')
+        except IOError as e:
+            if not self.get_presence(port_num):
+                return False # Port is not present, unable to set the eeprom
+            if port_num not in self.qsfp_ports:
+                return False # Read from eeprom only for QSFP ports
+            try:
+                eeprom = None
+                # Fill in write buffer
+                regval = 0x3 if lpmode else 0x1 # 0x3:Low Power Mode, 0x1:High Power Mode
+                buffer = create_string_buffer(1)
+                buffer[0] = chr(regval)
+                
+                # Write to eeprom
+                eeprom = open(self.port_to_eeprom_mapping[port_num], "r+b")
+                eeprom.seek(93)
+                eeprom.write(buffer[0])
+                return True
+            except IOError as e:
+                print "Error: unable to open file: %s" % str(e)
+                return False
+            finally:
+                if eeprom is not None:
+                    eeprom.close()
+                    time.sleep(0.01)
+        #raise NotImplementedError
+
+    def get_transceiver_change_event(self):
+        """
+        TODO: This function need to be implemented
+        when decide to support monitoring SFP(Xcvrd)
+        on this platform.
+        """
```

Comment:
When is the plan to implement this. So once PDDF is adapted change of events will be critical for run-time plug-in/out of media.